src/Classes/MoveController.cpp

```diff
@@ -53,13 +45,22 @@ void MoveController::initSprite(cocos2d::Vec2 center) {
   sprite->runAction(repeatAnimate);
 }
 
-cocos2d::Vec2 MoveController::findPlayerSpawn() const {
-  auto objectGroup = map->getObjectGroup("spawn point");
+Vec2 MoveController::findPlayerSpawn() const {
+  return findObjectPosition("spawn point", "spawn1");
+}
+
+Vec2 MoveController::findObjectPosition(const std::string layer, const std::string name) const {
+  auto objectGroup = map->getObjectGroup(layer);
   assert(objectGroup != nullptr);
-  auto sp = objectGroup->getObject("spawn1");
-  auto point = cocos2d::Vec2(sp["x"].asFloat(), sp["y"].asFloat());
-  assert(point != cocos2d::Vec2::ZERO);
-  return point;
+  auto sp = objectGroup->getObject(name);
+  auto point = Vec2(sp["x"].asFloat(), sp["y"].asFloat());
+  assert(point != Vec2::ZERO);
+  auto tileSize = map->getTileSize();
+  auto x0y0 = map->getLayer("water")->getTileAt(Vec2(0, 0))->getPosition();
+  auto translatedPoint = Vec2(point.x, x0y0.y - point.y);
+  auto pointToPixels = CC_POINT_POINTS_TO_PIXELS(translatedPoint);
+  auto gridPoint = Vec2(ceilf(pointToPixels.x / tileSize.width * 2.0f), ceilf(pointToPixels.y / tileSize.height));
+  return map->getLayer("water")->getTileAt(Vec2(gridPoint.x, gridPoint.y))->getPosition();
```

Comment:
Github shows comment right after intersting string (63). I suppose, that this line could be shorter. :)